docs/development/best_practices/rust.md

```diff
@@ -1,4 +1,87 @@
-# Rust Development
+## Rust Best Practices
```

Comment:
Main edits here are:

1. Mirroring the typescript.md
2. Shuffling the order of things, moving the "Tooling" section up and the "Learning" section down, to keep everything that SLI/EAC cares about grouped together
3. Updating the style guide permalink